lib/fastly_nsq/manager.rb

```diff
@@ -1,104 +1,80 @@
 # frozen_string_literal: true
 
-require 'set'
-
 class FastlyNsq::Manager
-  attr_reader :listeners
+  DEADLINE = 30
+  DEFAULT_POOL_SIZE = 5
+
+  attr_reader :done, :pool, :logger
 
-  def initialize(options = {})
-    @options = options
-    @done = false
-    @listeners = Set.new
-    @plock = Mutex.new
+  def initialize(logger: FastlyNsq.logger, **pool_options)
+    @done      = false
+    @logger    = logger
+    @pool      = FastlyNsq::PriorityThreadPool.new(
+      { fallback_policy: :caller_runs, max_threads: DEFAULT_POOL_SIZE }.merge(pool_options),
+    )
   end
 
-  def start
-    setup_configured_listeners
-    @listeners.each(&:start)
+  def topic_listeners
+    @topic_listeners ||= {}
   end
 
-  def quiet
-    return if @done
-    @done = true
+  def topics
+    topic_listeners.keys
+  end
 
-    FastlyNsq.logger.info { 'Terminating quiet listeners' }
-    @listeners.each(&:terminate)
+  def listeners
+    topic_listeners.values
```

Comment:
not a big deal, might be worth uniq'ing this or converting to a set.

Reply:
That's a good point.  Set might be easiest.